workflow/realisations.py

```diff
@@ -0,0 +1,412 @@
+        ): LAT_LON_DEPTH_SCHEMA,
+        Literal("strike", description="The strike bearing of the point source"): And(
+            float, is_valid_bearing
+        ),
+        Literal("dip", description="The dip angle of the point source"): And(
+            float, is_valid_bearing
+        ),
+        Literal(
+            "dip_dir", description="The dip direction bearing of the point source"
+        ): And(float, is_valid_bearing),
+    }
+)
+
+PLANE_SCHEMA = Schema(
+    {
+        Literal(
+            "type",
+            description="The type of the source geometry (Point, Plane or Fault)",
+        ): "plane",
+        Literal(
+            "corners",
+            description="The corners of the plane (shape 4 x 3: lat, lon, depth)",
+        ): And(Use(corners_to_array), is_correct_corner_shape, has_non_negative_depth),
+    }
+)
+
+FAULT_SCHEMA = Schema(
+    {
+        Literal(
+            "type",
+            description="The type of the source geometry (Point, Plane, or Fault)",
+        ): "fault",
+        Literal(
+            "corners",
+            description="The corners of the plane (shape 4n x 3: lat, lon, depth)",
+        ): And(
+            Use(corners_to_array),
+            Use(
+                lambda corners: corners.reshape(
+                    (-1, 4, 3), error="Corners cannot be reshaped to (n x 4 x 3)."
+                )
+            ),
+            has_non_negative_depth,
+        ),
+    }
+)
+
+SOURCE_SCHEMA = Schema({str: Or(POINT_SCHEMA, PLANE_SCHEMA, FAULT_SCHEMA)})
+
+
+SRF_GEN_SCHEMA = Schema(
+    {
+        Literal(
+            "genslip_dt",
+            description="The timestep for genslip (used to specify the resolution for the `TINIT` values)",
+        ): And(float, is_positive),
+        Literal("genslip_seed", description="The random seed passed to genslip"): And(
+            int, is_non_negative
+        ),
+        Literal("genslip_version", description="The version of genslip to use"): Or(
+            "5.4.2"
+        ),
+        Literal(
+            "srfgen_seed",
+            description="A second random seed for genslip (TODO: how does genslip use this value?)",
+        ): And(int, is_non_negative),
+    }
+)
+
+DOMAIN_PARAMETER_SCHEMA = Schema(
+    {
+        Literal("resolution", description="The simulation resolution (in km)"): And(
+            float, is_positive
+        ),
+        Literal(
+            "model_centroid", description="The centroid location of the model"
+        ): LAT_LON_SCHEMA,
+        Literal("model_width", description="The width of the model (in km)"): And(
+            float, is_positive
+        ),
+        Literal("model_length", description="The length of the model (in km)"): And(
+            float, is_positive
+        ),
+        Literal("model_depth", description="The depth of the model (in km)"): And(
+            float, is_positive
+        ),
+    }
+)
+
+REALISATION_SCHEMA = Schema(
```

Comment:
This schema is specific for type 5 only? Might be good to reflect this in the name?

Reply:
It's type-5 for now. The idea is to eventually port it to all types (hence the name). There is nothing specific to type-5 about the schema.

Reply:
The version of the realisation format is "5", but it could easily be 1.